src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -649,10 +671,24 @@ private async Task EntityMiddleware(DispatchMiddlewareContext dispatchContext, F
                         entityContext.ThrowInternalExceptionIfAny();
                         entityContext.ThrowApplicationExceptionsIfAny();
                     },
-#pragma warning restore CS0618
+                #pragma warning restore CS0618
                 },
                 CancellationToken.None);
 
+            if (result.ExecutionStatus == WrappedFunctionResult.FunctionResultStatus.FunctionsRuntimeError)

```

Comment:
I'm not sure I fully understand the implications of calling `SessionAbortedException` from inside of an Entity execution.

Reply:
Talked with Sebastian and it should be safe to abort Entity executions in the case of an exception.